vault/diagnose/storage_checks.go

```diff
@@ -80,3 +81,14 @@ func StorageEndToEndLatencyCheck(ctx context.Context, b physical.Backend) error
 	}
 	return nil
 }
+
+// ConsulDirectAccess verifies that consul is connecting to local agent,
+// versus directly to a remote server. We can only assume that the local address
+// is a server, not a client.
+func ConsulDirectAccess(config map[string]string) string {
+	serviceRegistrationAddr := config["address"]
+	if !strings.Contains(serviceRegistrationAddr, "localhost") && !strings.Contains(serviceRegistrationAddr, "127.0.0.1") {
+		return "consul storage does not connect to local agent, but directly to server"
```

Comment:
does it make sense to `const` this since we check for it in tests?

Reply:
Absolutely.